File: JavaTest/CallbackTest.java
import MyLibrary.*;
import org.junit.*;

import static org.junit.Assert.*;

public class CallbackTest extends TestBase {

    class MyCallback extends CallbackExample
    {
        public int ImplementMe()
        {
            return 54321;
        }
    }

    @Test
    public void OverrideTest() {
        MyCallback m = new MyCallback();
        assertEquals(54321, m.GetCallbackResult());
    }
}

File: CSharpTest/CallbackTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace CSharpTest
{
    class MyCallback : MyLibrary.CallbackExample
    {
        public override int ImplementMe()
        {
            return 54321;
        }
    };

    [TestClass]
    public class CallbackTest
    {
        [TestMethod]
        public void OverrideTest()
        {
            var m = new MyCallback();
            Assert.AreEqual(54321, m.GetCallbackResult());
        }
    }
}
